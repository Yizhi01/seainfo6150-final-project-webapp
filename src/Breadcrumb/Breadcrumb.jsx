import React from "react";
import { Link } from "react-router-dom";
import styles from "./Breadcrumb.module.css";
import PropTypes from "prop-types";

const Breadcrumb = (props) => {
  return (
    <div>
      <nav>
        <ol className={styles.breadcrumbList}>
          <li>
            <Link className={styles.breadcrumbTitle} to="/">Home</Link>
          </li>
          <li>
            <Link className={styles.breadcrumbTitle} to="/categoryList">Recipes</Link>
          </li>
          <li>
            <Link 
              className={styles.breadcrumbTitle}
              to={"/category/" + props.categoryId + "/" + props.categoryName}>
                {props.categoryName}
            </Link>
          </li>
        </ol>
      </nav>
    </div>
  );
};

Breadcrumb.propTypes = {
  categoryId: PropTypes.string.isRequired,
  categoryName: PropTypes.string.isRequired,
};

export default Breadcrumb;

