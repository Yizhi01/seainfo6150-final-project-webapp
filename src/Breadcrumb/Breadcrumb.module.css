.breadcrumbList {
  display: flex;
  list-style: none;
  font-size: 1.1em;
}

.breadcrumbList li+li:before {
  padding: 8px;
  color:green;
  content: ">";
}

.breadcrumbTitle {
  font-weight: 600;
  text-transform: uppercase;
  text-decoration: none;
  color: #393e46;
}

.breadcrumbTitle:hover {
  text-decoration: underline;
}