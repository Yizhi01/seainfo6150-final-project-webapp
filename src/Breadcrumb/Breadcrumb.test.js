import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Breadcrumb from "./Breadcrumb.jsx";

it("renders the Breadcrumb component correctly", () => {
  const { container } = render(
    <Router>
        <Breadcrumb categoryId="1" categoryName="Breakfast" />
    </Router>
  );
  expect(container).toMatchSnapshot();
});