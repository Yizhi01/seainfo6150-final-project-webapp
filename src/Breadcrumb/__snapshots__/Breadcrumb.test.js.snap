// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Breadcrumb component correctly 1`] = `
<div>
  <div>
    <nav>
      <ol
        class="breadcrumbList"
      >
        <li>
          <a
            class="breadcrumbTitle"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="breadcrumbTitle"
            href="/categoryList"
          >
            Recipes
          </a>
        </li>
        <li>
          <a
            class="breadcrumbTitle"
            href="/category/1/Breakfast"
          >
            Breakfast
          </a>
        </li>
      </ol>
    </nav>
  </div>
</div>
`;
